connector/auth/jwt_token_validation.go

```diff
@@ -60,26 +60,34 @@ func (jv JwtTokenValidator) AuthenticateRequest(ctx context.Context, activity sc
 	return identity, nil
 }
 
-func (jv JwtTokenValidator) getIdentity(authHeader string) (ClaimsIdentity, error) {
-
-	jwksURL, err := jv.getJwkURL(metadataURL)
-	if err != nil {
-		return nil, err
-	}
+func (jv *JwtTokenValidator) getIdentity(authHeader string) (ClaimsIdentity, error) {
 
 	getKey := func(token *jwt.Token) (interface{}, error) {
 
-		set, err := jwk.FetchHTTP(jwksURL)
-		if err != nil {
-			return nil, err
+		if jv.jwkCache.IsExpired() {
```

Comment:
Please add one-liner comments about what we are doing here

Reply:
Added a comment.